docs/en/query_language/select.md

```diff
@@ -957,12 +957,31 @@ Running a query may use more memory than 'max_bytes_before_external_sort'. For t
 
 External sorting works much less effectively than sorting in RAM.
 
-### SELECT Clause
+### SELECT Clause {#select-select}
 
-The expressions specified in the SELECT clause are analyzed after the calculations for all the clauses listed above are completed.
-More specifically, expressions are analyzed that are above the aggregate functions, if there are any aggregate functions.
-The aggregate functions and everything below them are calculated during aggregation (GROUP BY).
-These expressions work as if they are applied to separate rows in the result.
+[Expressions](syntax.md#syntax-expressions) that specified in the `SELECT` clause are analyzed after the calculations for all the clauses listed above are completed. More specifically, expressions are analyzed that are above the aggregate functions, if there are any aggregate functions. The aggregate functions and everything below them are calculated during aggregation (`GROUP BY`). These expressions work as if they are applied to separate rows in the result.
+
+If you want to get all columns in the result, use the asterisk (`*`) symbol. For example, `SELECT * FROM ...`.
+
+You can also use the `COLUMNS` expression to match some columns in the result by a regular expression.
+
+```sql
+COLUMNS('regexp')
+```
+
+Example:
+
+```sql
+SELECT COLUMNS('n') + COLUMNS('u') FROM system.numbers LIMIT 2
```

Comment:
* what will happen if no columns match?
* what will happen if multiple columns match? does it matter if all have same/compatible type?
* what will happen if we replace `+` with `,` in this example?
* what will be the result of `toTypeName(COLUMNS(...))`?

Reply:
@mfridental Can you formulate some limitations for using this functionality that can help to answer Ivan's questions?

Reply:
> what will happen if no columns match?

The function `+` will be called with zero number of arguments and exception will be thrown.

> what will happen if multiple columns match?

The function `+` will be called with more than two arguments and exception will be thrown.

> what will happen if we replace + with , in this example?

Everything gonna be alright.

> what will be the result of toTypeName(COLUMNS(...))

The only valid way is when COLUMNS will match single column and the function `toTypeName` will be called with a single argument.